synthesizer/process/src/lib.rs

```diff
@@ -131,20 +167,50 @@ impl<N: Network> Process<N> {
     /// Adds a new stack to the process.
     /// If you intend to `execute` the program, use `deploy` and `finalize_deployment` instead.
     #[inline]
-    pub fn add_stack(&mut self, stack: Stack<N>) {
+    pub fn add_stack(&self, stack: Stack<N>) {
         // Add the stack to the process.
-        self.stacks.insert(*stack.program_id(), Arc::new(stack));
+        let mut lock = self.stacks.lock();
+        match lock {
+            Ok(ref mut lock) => lock.put(*stack.program_id(), Arc::new(stack)),
+            Err(_) => None,
+        };
     }
 }
 
 impl<N: Network> Process<N> {
+    /// Initializes a new process.
+    /// Assumption: this is only called in test code.
+    #[inline]
+    pub fn load_testing_only() -> Result<Self> {
+        Process::load_from_storage(Some(aleo_std::StorageMode::Development(0)))
+    }
+
     /// Initializes a new process.
     #[inline]
-    pub fn load() -> Result<Self> {
-        let timer = timer!("Process::load");
+    pub fn load_from_storage(storage_mode: Option<StorageMode>) -> Result<Self> {
+        let timer = timer!("Process::load_testing_only");
```

Comment:
shouldn't this be `Process::load_from_storage`? alternatively, this could be moved to the beginning of `load_testing_only`

Reply:
Good catch 01d9a0b49